### PR TITLE
chore: Bump rust dependencies

### PR DESCRIPTION
**Pull request by NickLarsenNZ**
# Description

Bump operator-rs to 0.70.0 and other deps updated.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [x] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

### CODE REVIEW

**Comment by NickLarsenNZ** [comment]
nightly tests pass

```
❯ ./scripts/run-tests --test-suite nightly
...
--- PASS: kuttl (219.43s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/delete-rolegroup_zookeeper-3.9.2_openshift-false (146.85s)
        --- PASS: kuttl/harness/smoke_zookeeper-3.9.2_use-server-tls-true_use-client-auth-tls-true_openshift-false (153.51s)
        --- PASS: kuttl/harness/znode_zookeeper-latest-3.9.2_openshift-false (26.66s)
        --- PASS: kuttl/harness/logging_zookeeper-3.9.2_openshift-false (65.01s)
        --- PASS: kuttl/harness/cluster-operation_zookeeper-latest-3.9.2_openshift-false (39.26s)
PASS
```

File: Cargo.toml [hunk 1]
```diff
@@ -17,16 +17,16 @@ fnv = "1.0"
 futures = { version = "0.3", features = ["compat"] }
 indoc = "2.0"
 pin-project = "1.1"
-rstest = "0.19"
+rstest = "0.21.0"
```
Comment (Techassi) [inline Cargo.toml:20]: nit: Don't we usually just use X.Y? So `0.21` in this case?

**Review by Techassi — COMMENTED** [review]
LGTM, one minor nit.

File: Cargo.toml (hunk 1 above)
Comment (NickLarsenNZ) [reply Cargo.toml]: Good catch, done in 4ece0ede373e4e2abc98456caa6eda141622e89b

**Review by Techassi — APPROVED** [review]

**Pull request merged**